include/clang/Tooling/Refactor/IndexerQuery.h

```diff
@@ -181,6 +181,33 @@ class DeclPredicateNotPredicate : public DeclPredicateNode {
 
 } // end namespace detail
 
+enum class QueryBoolResult {
+  Unknown,
+  Yes,
+  No,
+};
+
+// FIXME: Either PersistentDeclRef or Decl *
+template <typename T> struct Indexed {
+  T Decl;
+  // FIXME: Generalize better in the new refactoring engine.
+  QueryBoolResult BoolResult = QueryBoolResult::Unknown;
+
+  Indexed(T Decl, QueryBoolResult BoolResult = QueryBoolResult::Unknown)
```

Comment:
It doesn't look like we take advantage of the fact that BoolResult has a default value. Maybe it should be dropped?